modules/core/client/services/auth.service.client.js

```diff
@@ -1,66 +1,82 @@
 'use strict';
 
 angular.module('tropicalbs')
-  .factory('Auth', ['$http', '$location', '$window', '$cookies', 'User', function ($http, $location, $window, $cookies, User) {
+  .factory('authService', authService);
 
-    var auth = {};
+authService.$inject =  ['$cookies', '$http', '$location', '$window', 'userService'];
 
-    auth.login = function (user) {
-      return $http({
-        method: 'POST',
-        url: 'api/core/users/login',
-        data: user
-      })
-      .then(function (res) {
-        if (res.secure) {
-          $cookies.put('userToken', res.data.token, {secure: true});
-        } else {
-          $cookies.put('userToken', res.data.token);
-        }
-        User.logIn(res.data.user);
-        return res;
-      });
-    };
+function authService ($cookies, $http, $location, $window, userService) {
 
-    auth.signup = function (user) {
-      return $http({
-        method: 'POST',
-        url: 'api/core/users/signup',
-        data: user
-      })
-      .then(function (res) {
-        if (res.secure) {
-          $cookies.put('userToken', res.data.token, {secure: true});
-        } else {
-          $cookies.put('userToken', res.data.token);
-        }
-        User.logIn(res.data.user);
-        return res;
-      });
-    };
+  var auth = {
+    checkAuth: checkAuth,
+    login: login,
+    logout: logout,
+    signup: signup
+  };
 
-    auth.logout = function () {
-      $cookies.remove('userToken');
-      User.setDefault();
-      $window.location.reload();
-    };
+  return auth;
 
-    auth.checkAuth = function () {
-      return $http({
-        method: 'POST',
-        url: 'api/core/users/checkauth'
-      })
-      .then(function (res) {
-        User.logIn(res.data.user);
-        return res;
-      })
-      .catch(function (e) {
-        console.log(e);
-        //if AUTH is unsuccessful, 'logout'
-        auth.logout();
-      });
-    };
+  //////////
 
-    return auth;
+  function checkAuth () {
+    return $http({
+      method: 'POST',
+      url: 'api/core/users/checkauth'
+    })
+    .then(userLogin)
+    .catch(handleAuthError);
 
```

Comment:
I don't like this inconsistency on spacing. .then should be indented.

To make it look pretty, move the object to it's own variable:

```
var req = {
  method: 'POST',
  url: 'api/core/users/checkauth'
};

return $http(req)
  .then(userLogin)
  .catch(handleAuthError);